src/scores/processing.py

```diff
@@ -1,8 +1,188 @@
 """Tools for processing data for verification"""
+import operator
 from typing import Union
 
+import numpy as np
+import pandas as pd
 import xarray as xr
 
+INEQUALITY_MODES = {
+    ">=": (operator.ge, -1),
+    ">": (operator.gt, 1),
+    "<=": (operator.le, 1),
+    "<": (operator.lt, -1),
+}
+# '==' does not map to `operater.eq` and '!=' does not map to operator.ne on purpose.
+# This is because we wish to test within a tolerance.
+EQUALITY_MODES = {"==": (operator.le), "!=": (operator.gt)}
+
+
+def check_binary(data: Union[xr.DataArray, xr.Dataset], name: str):
+    """
+    Checks that data does not have any non-NaN values out of the set {0, 1}
+
+    Args:
+        data: The data to convert to check if only contains binary values
+    Raises:
+        ValueError: if there are values in `fcst` and `obs` that are not in the
+            set {0, 1, np.nan} and `check_args` is true.
+    """
+    unique_values = pd.unique(data.values.flatten())
+    unique_values = unique_values[~np.isnan(unique_values)]
+    binary_set = {0, 1}
+
+    if not set(unique_values).issubset(binary_set):
+        raise ValueError(f"`{name}` contains values that are not in the set {{0, 1, np.nan}}")
+
+
+def comparative_discretise(data, comparison, mode, abs_tolerance=None):
```

Comment:
Based on the docs, data could be type hinted with XarrayLike

Reply:
updated